entities/src/main/java/com/flipkart/varadhi/entities/TopicResource.java

```diff
@@ -5,8 +5,8 @@
 
 @Value
 @EqualsAndHashCode(callSuper = true)
-@ValidateVaradhiResource(message = "Invalid Topic name. Check naming constraints.", max = 64)
-public class TopicResource extends VaradhiResource {
+@ValidateMetaStoreEntity(message = "Invalid Topic name. Check naming constraints.", max = 64)
+public class TopicResource extends MetaStoreEntity {
```

Comment:
shouldn't this extend VersionedEntity as it is user facing entity?

Reply:
Topic refactoring to conform to this model will happen as separate PR. I think that PR is being already worked on.